pkg/resources/provisioning.cattle.io/v1/cluster/validator.go

```diff
@@ -697,6 +703,51 @@ func (p *provisioningAdmitter) validatePodDisruptionBudget(oldCluster, cluster *
 	return admission.ResponseAllowed(), nil
 }
 
+// validateS3Secret checks if the S3 cloud credential secret referenced in the cluster exists.
+func (p *provisioningAdmitter) validateS3Secret(oldCluster, cluster *v1.Cluster) (*admissionv1.AdmissionResponse, error) {
+	if cluster.Name == localCluster {
```

Comment:
I think you need a check here `if oldCluster == nil`, so that you still validate on create, otherwise it will likely panic.